### PR TITLE
fwupdmgr: Hide devices that aren't updatable by default

### PR DESCRIPTION
**Pull request by superm1**
They are hidden from get-devices, get-updates, and get-topology
They can be viewed by adding '--show-all-devices' to command line

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -62,6 +62,7 @@ typedef struct {
 	gboolean		 no_reboot_check;
 	gboolean		 no_unreported_check;
 	gboolean		 assume_yes;
+	gboolean		show_all_devices;
```
Comment (hughsie) [inline src/fu-util.c]: needs a space after the tab.

File: src/fu-util.c [hunk 2]
```diff
@@ -573,6 +574,28 @@ fu_util_print_device_tree (GNode *n, gpointer data)
...
+fu_util_remove_extra_devices (FuUtilPrivate *priv, GPtrArray *devs, GError **error)
+{
+	if (priv->show_all_devices)
+		return TRUE;
+
+	for (guint i = 0; i < devs->len; i++) {
+		FwupdDevice *dev_tmp = g_ptr_array_index (devs, i);
+		if (fwupd_device_has_flag (dev_tmp, FWUPD_DEVICE_FLAG_UPDATABLE))
+			continue;
+		if (!g_ptr_array_remove (devs, dev_tmp)) {
```
Comment (hughsie) [inline src/fu-util.c]: I don't think it's safe to carry on iterating the list when you've removed an element. Normally you just return a new list which has only the elements you require added to it.

File: src/fu-util.c (hunk 1 above)
Comment (superm1) [reply src/fu-util.c]: Fixed

File: src/fu-util.c (hunk 2 above)
Comment (superm1) [reply src/fu-util.c]: Ah, I thought that glib was abstracting this risk, but it's not.  I've moved it to just do the check in place for get-devices and get-topology.  get-updates implicitly doesn't benefit from it.

File: src/fu-util.c [hunk 3]
```diff
@@ -590,7 +593,7 @@ fu_util_get_topology (FuUtilPrivate *priv, gchar **values, GError **error)
 		g_print ("%s\n", _("No hardware detected with firmware update capability"));
 		return TRUE;
 	}
-	fu_util_build_device_tree (root, devs, NULL);
+	fu_util_build_device_tree (root, devs, priv->show_all_devices, NULL);
```
Comment (hughsie) [inline src/fu-util.c]: Maybe just pass self to `fu_util_build_device_tree`?

File: src/fu-util.c [hunk 4]
```diff
@@ -2353,6 +2358,9 @@ main (int argc, char *argv[])
 		{ "no-reboot-check", '\0', 0, G_OPTION_ARG_NONE, &priv->no_reboot_check,
 			/* TRANSLATORS: command line option */
 			_("Do not check for reboot after update"), NULL },
+		{ "show-all-devices", '\0', 0, G_OPTION_ARG_NONE, &priv->show_all_devices,
+			/* TRANSLATORS: command line option */
+			_("Show devices that aren't updatable"), NULL },
```
Comment (hughsie) [inline src/fu-util.c]: aren't -> "are not" pls

File: src/fu-util.c (hunk 3 above)
Comment (superm1) [reply src/fu-util.c]: Sure, switched around.

File: src/fu-util.c (hunk 4 above)
Comment (superm1) [reply src/fu-util.c]: fixed

**Comment by hughsie** [comment]
Lgtm, thanks.

**Pull request merged**